### ISSUE TITLE
deploy options dropped running docker-compose config

### ISSUE THREAD

**Issue by kinghuang** [docker/compose#4517]
I'm having a strange problem running `docker-compose config` on a project. I use the `config` command to combine multiple Compose files together for deployment using `docker stack deploy --compose-file …`. For some reason, when I run `config` on this project, Docker Compose is dropping the `deploy` options on the services.

[monitoring.zip](https://github.com/docker/compose/files/790739/monitoring.zip)

The monitoring project consists of three Compose files, specified in `.env`. Running `docker-compose config` in this directory produces a config with the `deploy` options missing.

I'm using Docker Compose 1.11.1.

```
→  monitoring git:(monitoring-stack) ✗ docker-compose --version
docker-compose version 1.11.1, build 7c5d5e4
```

**Comment by shin-** [comment]
Probably due to the fact that those options are ignored by `docker-compose`, but it's worth looking into a fix. Thanks for the report.

**Comment by kinghuang** [comment]
Thanks, @shin-. It seems like it might be dropping `deploy` only when there are multiple Compose files. If you run docker-compose with just the `exporters.yml` file, for example, the `deploy` options stay.

**Comment by bmitch3020** [comment]
I'm encountering the same in my environment. We're trying to migrate from `docker-compose` to `docker stack deploy` with multiple docker-compose.*.yml files and variables and need to use the `config` command to reprocess the inputs into a single yml.

**Comment by kinghuang** [comment]
@bmitch3020 Yeah, it's been an ongoing nuisance for me. I did trace down the problem to the `merge_service_dicts()` function not handling the `deploy` key. I'll take a stab at a fix, later.

**Comment by kinghuang** [comment]
Fixed in PR #4669. @shin- PTAL?
